dockerscripts/healthcheck.go

```diff
@@ -0,0 +1,144 @@
+/*
+ * Minio Cloud Storage, (C) 2019 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+const (
+	initGraceTime = 300
+	healthPath    = "/minio/health/live"
+	scheme        = "http://"
+	schemeSecure  = "https://"
+)
+
+// returns container boot time by finding
+// modtime of /proc/1 directory
+func getStartTime() time.Time {
+	di, err := os.Stat("/proc/1")
+	if err != nil {
+		// Cant stat proc dir successfully, exit with error
+		os.Exit(1)
+	}
+	return di.ModTime()
+}
+
+// Finds the ip/port of the Minio process
+// running in the container
+func findEndpoint() string {
+	op, err := exec.Command("netstat", "-ntlp").Output()
+	if err != nil {
+		// netstat unsuccessful, exit with error
+		os.Exit(1)
+	}
+	// split netstat output in rows
+	lines := strings.Split(string(op), "\n")
```

Comment:
You can instead use bufio.Scanner here to look at line by line combine this with the os.Stdin reader from exec.Command() is more idiomatic style.